frontend/javascripts/maintenance_banner.tsx

```diff
@@ -0,0 +1,106 @@
+import {
+  MaintenanceInfo,
+  listCurrentAndUpcomingMaintenances,
+  updateNovelUserExperienceInfos,
+} from "admin/admin_rest_api";
+import { Alert } from "antd";
+import FormattedDate from "components/formatted_date";
+import { useInterval } from "libs/react_helpers";
+import { setActiveUserAction } from "oxalis/model/actions/user_actions";
+import { Store } from "oxalis/singletons";
+import React, { useState } from "react";
+
+export function MaintenanceBanner() {
+  const { activeUser } = Store.getState();
+  const [currentAndUpcomingMaintenances, setcurrentAndUpcomingMaintenances] = useState<
+    Array<MaintenanceInfo>
+  >([]);
+  const [position, setPosition] = useState<Object>({ top: 48 });
+  const [isTop, setIsTop] = useState(true);
+  useInterval(
+    async () => {
+      if (activeUser == null) return;
+      setcurrentAndUpcomingMaintenances(await listCurrentAndUpcomingMaintenances());
+    },
+    10000, //TODO find value for production
+  );
+  if (activeUser == null) return <></>;
+  const activeUsersLatestAcknowledgedMaintenance =
+    activeUser?.novelUserExperienceInfos.latestAcknowledgedMaintenanceInfo;
+
+  const saveUserClosedMaintenanceInfo = (closestUpcomingMaintenance: MaintenanceInfo) => {
+    if (activeUser == null) return;
+    const [nextMaintenanceAcknowledged] = updateNovelUserExperienceInfos(activeUser, {
+      latestAcknowledgedMaintenanceInfo: closestUpcomingMaintenance.id,
+    });
+    Store.dispatch(setActiveUserAction(nextMaintenanceAcknowledged));
+  };
+
+  const toggleTopOrBottomPosition = () => {
+    setPosition(isTop ? { top: "48px" } : { bottom: "20px" });
```

Comment:
TODO: unify style to write position

Reply:
1) it's not necessary to pass a string with px. you can just pass a number
2) use variables for both values